moveit_experimental/moveit_jog_arm/src/low_pass_filter.cpp

```diff
@@ -38,15 +38,46 @@
  */
 
 #include <moveit_jog_arm/low_pass_filter.h>
+#include <cmath>
 #include <string>
+#include <ros/ros.h>
 
-static const std::string LOGNAME = "low_pass_filter";
+const std::string LOGNAME = "low_pass_filter";
+constexpr double EPSILON = 1e-9;
 
 namespace moveit_jog_arm
 {
 LowPassFilter::LowPassFilter(double low_pass_filter_coeff)
+  : previous_measurements_{ 0., 0. }
+  , previous_filtered_measurement_(0.)
+  , filter_coeff_(low_pass_filter_coeff)
+  , scale_term_(1. / (1. + low_pass_filter_coeff))
+  , feedback_term_(-low_pass_filter_coeff + 1.)
 {
-  filter_coeff_ = low_pass_filter_coeff;
+  // guarentee this doesn't change because the logic depends on this length implicity
+  static_assert(LowPassFilter::FILTER_LENGTH == 2);
+
+  if (std::abs(feedback_term_) < EPSILON)
+  {
+    ROS_WARN_STREAM_NAMED(
+        LOGNAME, "Filter coefficient value of "
+                     << filter_coeff_
+                     << " resulted in feedback term of 0. "
+                        " This results in a window averaging Finite Impulse Response (FIR) filter with a gain of "
+                     << scale_term_ * LowPassFilter::FILTER_LENGTH);
+  }
+  if (std::isinf(scale_term_))
+  {
+    ROS_FATAL_STREAM_NAMED(LOGNAME, "Filter coefficient value of "
+                                        << filter_coeff_
+                                        << ", outputs from filter will be inf because denominator of scale is 0");
+  }
+  if (filter_coeff_ < 1.)
+  {
+    ROS_ERROR_STREAM_NAMED(
+        LOGNAME,
+        "Filter coefficient < 1. makes single order Butterworth an unstable Infinite Impulse Response (IIR) filter, ");
```

Comment:
Delete the ", " at the end of this line.

I think this error message might be too detailed for most. How about:

"Filter coefficient < 1. makes the lowpass filter unstable"

Reply:
done